lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -2521,10 +2529,54 @@ private void updateSuffixTextVisibility() {
     int oldSuffixVisibility = suffixTextView.getVisibility();
     boolean visible = suffixText != null && !isHintExpanded();
     suffixTextView.setVisibility(visible ? VISIBLE : GONE);
+    updateEndLayoutVisible();
     if (oldSuffixVisibility != suffixTextView.getVisibility()) {
       getEndIconDelegate().onSuffixVisibilityChanged(visible);
     }
     updateDummyDrawables();
+    updateEndLayoutVisible();
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  private void updateEndLayoutVisible() {
+    if (endIconView == null || endIconFrame == null || errorIconView == null
+        || suffixTextView == null || endLayout == null)
+      return;;
+
+    if (!hasEndIcon() && endIconView.getVisibility() != GONE)
+      endIconView.setVisibility(GONE);
+
+    if (endIconFrame.getVisibility() != endIconView.getVisibility())
+      endIconFrame.setVisibility(endIconView.getVisibility());
+
+    int endIconFrameVis = endIconFrame.getVisibility();
```

Comment:
Don't use uncommon abbreviations, rest the same.

Reply:
Give example

Reply:
Please replace vis with visibility